source/slang/check.cpp

```diff
@@ -6049,7 +6051,7 @@ namespace Slang
             // through types `X` that are also builtin scalar types.
             //
             RefPtr<Type> bestType;
-            if(auto basicType = type.As<BasicExpressionType>())
+            if(auto basicType = type.dynamicCast<BasicExpressionType>())
```

Comment:
This is another case where I suspect the original code should have been using a canonical cast, so that `as` is appropriate now.